includes/embeds/class-amp-tiktok-embed-handler.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Class AMP_Tiktok_Embed_Handler
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Tiktok_Embed_Handler
+ */
+class AMP_Tiktok_Embed_Handler extends AMP_Base_Embed_Handler {
+
+	/**
+	 * Registers embed.
+	 */
+	public function register_embed() {
+		add_filter( 'embed_oembed_html', [ $this, 'filter_embed_oembed_html' ], 10, 3 );
+	}
+
+	/**
+	 * Unregisters embed.
+	 */
+	public function unregister_embed() {
+		remove_filter( 'embed_oembed_html', [ $this, 'filter_embed_oembed_html' ], 10 );
+	}
+
+	/**
+	 * Filter oEmbed HTML for Tiktok to prepare it for AMP.
+	 *
+	 * @param mixed  $html The oEmbed HTML.
+	 * @param string $url The attempted embed URL.
+	 * @param array  $attr Attributes.
+	 * @return string Embed.
+	 */
+	public function filter_embed_oembed_html( $html, $url, $attr ) {
+		$parsed_url = wp_parse_url( $url );
+
+		if ( false === strpos( $parsed_url['host'], 'tiktok.com' ) ) {
+			return $html;
+		}
+
+		$video_id = $this->get_video_id( $html );
+
+		if ( ! $video_id ) {
+			return AMP_HTML_Utils::build_tag(
+				'a',
+				[
+					'href'  => esc_url_raw( $url ),
+					'class' => 'amp-wp-embed-fallback',
+				],
+				esc_html( $url )
+			);
+		}
+
+		if ( $video_id && preg_match( '#<blockquote(?:\s.*?)?>(?P<content>.*?)</blockquote>#s', $html, $matches ) ) {
+			$content = str_replace( '<section>', '<section placeholder>', $matches['content'] );
+
+			$iframe = AMP_HTML_Utils::build_tag(
+				'amp-iframe',
+				[
+					'width'   => $attr['width'],
+					'height'  => $attr['height'],
+
+					/*
+					 * A `lang` query parameter is added to the URL via JS. This can't be determined here so it is not
+					 * added. Whether it alters the embed in any way or not has not been determined.
+					 */
+					'src'     => 'https://www.tiktok.com/embed/v2/' . $video_id,
```

Comment:
Want to point out a potential issue here; the `lang` query parameter is added by the accompanied JS and so cannot be determined here. I'm not sure if this affects the embed in any way.

Reply:
One possibility would be to make use of a URL substitution. 

```suggestion
					/*
					 * A `lang` query parameter is added to the URL via JS. This can't be determined here so it is not
					 * added. Whether it alters the embed in any way or not has not been determined.
					 */
					'src'     => 'https://www.tiktok.com/embed/v2/' . $video_id . '?lang=BROWSER_LANGUAGE',
```

See https://github.com/ampproject/amphtml/blob/master/spec/amp-var-substitutions.md

_But_ it seems AMP doesn't support this yet, as this PR was closed: https://github.com/ampproject/amphtml/pull/10884

Reply:
Right, I haven't found any definite proof of URL substitution being done on the `src` attribute of an `amp-iframe`.